### PR TITLE
improve codebase and performance

### PR DESCRIPTION
**Pull request by Uzlopak**
before:

```
  no header   x 9,918,143 ops/sec ±0.41% (193 runs sampled)
  1 address   x 6,509,341 ops/sec ±0.59% (190 runs sampled)
  2 addresses x 3,776,782 ops/sec ±1.24% (188 runs sampled)
  5 addresses x 2,228,356 ops/sec ±0.28% (181 runs sampled)
```

after:
```
no header   x 1,246,115,625 ops/sec ±0.24% (196 runs sampled)
  1 address   x    66,139,899 ops/sec ±0.91% (193 runs sampled)
  2 addresses x    10,107,089 ops/sec ±0.34% (195 runs sampled)
  5 addresses x     4,374,716 ops/sec ±0.48% (194 runs sampled)
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,56 +14,40 @@ module.exports = forwarded
...
 
-function forwarded (req) {
+module.exports = function forwarded (req) {
   if (!req) {
     throw new TypeError('argument req is required')
   }
 
   // simple header parsing
-  const proxyAddrs = parse(req.headers['x-forwarded-for'] || '')
   const socketAddr = req.socket.remoteAddress
```
Comment (Uzlopak) [inline index.js:18]: Maybe it gets some yoctoseconds faster if i dont use the variable and directly use it in the code below.

**Comment by Uzlopak** [comment]
I was actually investigating if i could improve the original package of dougwilson. Then i saw you already created a fork. I hope you like it...

**Comment by Uzlopak** [comment]
@Fdawgs 
@mcollina 
@Eomm 
@kibertoad

**Review by Fdawgs — APPROVED** [review]
Disgusting, lgtm

**Comment by Uzlopak** [comment]
@Fdawgs 

Thank you xD. I added some unit tests for some edge cases, just to be on the safe side. Please just run the workflows again ;)

I even had a faster implementation, but that would have been a security risk, as it would use recursion. And Forwarded-For is a security header, so I didnt wanted that somebody could raise a MaxStackCall Error and poitentially crashing the service.
